components/admin_console/admin_definition.jsx

```diff
@@ -401,11 +420,11 @@ const AdminDefinition = {
         sectionTitle: t('admin.sidebar.userManagement'),
         sectionTitleDefault: 'User Management',
         isHidden: it.all(
-            it.not(it.userHasReadPermissionOnResource('user_management.users')),
-            it.not(it.userHasReadPermissionOnResource('user_management.groups')),
-            it.not(it.userHasReadPermissionOnResource('user_management.teams')),
-            it.not(it.userHasReadPermissionOnResource('user_management.channels')),
-            it.not(it.userHasReadPermissionOnResource('user_management.permissions')),
+            it.not(it.userHasReadPermissionOnResource(RESOURCE_KEYS.USER_MANAGEMENT.USERS)),
+            it.not(it.userHasReadPermissionOnResource(RESOURCE_KEYS.USER_MANAGEMENT.GROUPS)),
+            it.not(it.userHasReadPermissionOnResource(RESOURCE_KEYS.USER_MANAGEMENT.TEAMS)),
+            it.not(it.userHasReadPermissionOnResource(RESOURCE_KEYS.USER_MANAGEMENT.CHANNELS)),
+            it.not(it.userHasReadPermissionOnResource(RESOURCE_KEYS.USER_MANAGEMENT.PERMISSIONS)),
```

Comment:
It would be great if these arrays of sub-permissions updated these parent sections from a single source data structure to make it impossible to forget to add one. 

Actually, maybe we should switch to a `permissions` parameter that handled the imperative part of the logic (namely, disabling and or hiding). That way a developer making a change to these menu items later doesn't have to think about any of `not`, `all`, `isDisabled`, `isHidden`, and `userHasReadPermissionOnResource` things.

Reply:
@hahmadia as we discussed offline, I created a ticket for us to handle this refactor: https://mattermost.atlassian.net/plugins/servlet/mobile?originPath=%2Fbrowse%2FMM-34281#issue/MM-34281